src/java.base/unix/classes/sun/nio/fs/UnixFileAttributeViews.java

```diff
@@ -72,38 +72,23 @@ public void setTimes(FileTime lastModifiedTime,
             // permission check
             file.checkWrite();
 
+            // use a file descriptor if possible to avoid a race due to
+            // accessing a path more than once as the file at that path could
+            // change.
+            // if path is a symlink, then the open should fail with ELOOP and
+            // the path will be used instead of the file descriptor.
             boolean haveFd = false;
```

Comment:
In passing, we don't need haveFd, instead we can use fd >= 0 as is done in other places.

Reply:
Yes, I had thought of ding that but left `haveFd` as it had been there before. Will change.

Reply:
`s/haveFd/fd >= 0/` in fc5f8e8.